packages/crayons-core/src/components/datepicker/datepicker.tsx

```diff
@@ -850,24 +822,116 @@ export class Datepicker {
       );
       this.value = `${formattedFromDate} to ${formattedToDate}`;
     }
-    if (
-      (this.maxYear && Number(this.year) > this.maxYear) ||
-      (this.minYear && Number(this.year) < this.minYear)
-    ) {
-      if (this.value) {
+  }
+
+  setDateAndErrorState(checkDate = false) {
+    this.isDateInvalid =
+      checkDate && !this.isDateWithinMinMaxDate(parseISO(this.value).valueOf())
```

Comment:
I think we don't need a ternary operator here. `checkDate && ...` should anyway be returning a boolean value only.

Reply:
fixed.